src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -734,17 +736,7 @@ async Task<bool> NextResultSchemaOnly(bool async, bool isConsuming = false, Canc
 
                 // Found a resultset
                 if (RowDescription is not null)
-                {
```

Comment:
I don't have much context over this, but do you remember why we need this in the regular path but not here in the SchemaOnly path?

Reply:
We only need `ColumnInfoCache` while reading columns, and since with `SchemaOnly` we do not have a result set, there is no reason for us to even touch the cache.